backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxMetricsApi.scala

```diff
@@ -38,13 +39,29 @@ class VeloxMetricsApi extends MetricsApi with Logging {
   }
 
   override def genInputIteratorTransformerMetrics(
+      child: SparkPlan,
       sparkContext: SparkContext): Map[String, SQLMetric] = {
+    def metricsPlan(plan: SparkPlan): SparkPlan = {
+      plan match {
+        case ColumnarInputAdapter(child) => metricsPlan(child)
+        case q: QueryStageExec => metricsPlan(q.plan)
+        case _ => plan
+      }
+    }
```

Comment:
Can we only collect child's metrics when it's a broadcast exchange? Thanks.

Reply:
Hi @zhztheplayer , can you please clarify about the intention a little bit? If we only collect child's metrics for broadcast exchange, there'll be no output metrics(`numOutputRows` and `outputVectors`) for other operators then.

Reply:
I meant we can still keep the current code path in main stream code for the case of `child != broadcast exchange`. When `child == broadcast exchange`, we can grab the metrics from `child.metrics` to avoid duplicating the numbers. Does that sound feasible to you?

Reply:
> I meant we can still keep the current code path in main stream code for the case of `child != broadcast exchange`. When `child == broadcast exchange`, we can grab the metrics from `child.metrics` to avoid duplicating the numbers. Does that sound feasible to you?

Sure, I am fine with it. Will update.